File: Data Structures & Algorithms/is-anagram/submission-1.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
         if (s.size() != t.size()) return false; // Early exit if lengths differ

    std::unordered_map<char, int> freq;
    
    // Increment counts for 's'
    for (char c : s) freq[c]++;
    
    // Decrement counts for 't' and check for mismatches
    for (char c : t) {
        if (--freq[c] < 0) return false; // More 'c' in 't' than 's'
    }
    
    return true; // All counts balanced (no negative values)
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        freq = {}
        for i in nums:
            if i in freq:
                freq[i]+=1
                if(freq[i]==2):
                    return True
            else:
                freq[i] = 1
        return False